### PR TITLE
Mdas 459 implementierung s3 eai endpunkt loeschkonzept

### PR DESCRIPTION
**Pull request by sfi2022**
### Description
Loeschkonzept hinzugefuegt.
### Reference

Issues:

### Definition of Done
*Advice: Unchecked checkboxes are not accepted by the build pipeline. Cross out any non-relevant item by using '~~'.*

- [x] Acceptance criteria are met
- [x] Build/Test workflow has successfully finished
- [x] Release notes are complemented
- [x] Documentation is complemented (operator manual, system specification)
~~- [ ] Frontend is locally smoke-tested~~
- [x] Board is updated
- [x] Infrastructure is adjusted

Please delete all side branches after merging.

### CODE REVIEW

File: src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java [hunk 1]
```diff
@@ -33,15 +34,25 @@ public void configure() {
                 .handled(true)
                 .process(exchange -> {
                     var s3Exception = exchange.getProperty(Exchange.EXCEPTION_CAUGHT, S3Exception.class);
-                    log.error("Error occurred in route", s3Exception);
+                    if (exchange.getIn().getHeader(Constants.CAMEL_SERVLET_CONTEXT_PATH).equals(Constants.ARCHIVE_ENTITY)) {
```
Comment (martind260) [inline src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java:37]: Das kann man in eine Methode "logException" auslagern.

File: src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java [hunk 2]
```diff
@@ -33,15 +34,25 @@ public void configure() {
...
+                    }
                     exchange.getMessage().setBody(ErrorResponseBuilder.build(s3Exception.statusCode(), s3Exception.getClass().getName()));
                 });
 
         onException(ResolveEndpointFailedException.class)
                 .handled(true)
                 .process(exchange -> {
                     var s3Exception = exchange.getProperty(Exchange.EXCEPTION_CAUGHT, ResolveEndpointFailedException.class);
-                    log.error("Error occurred in route", s3Exception);
+                    if (exchange.getIn().getHeader(Constants.CAMEL_SERVLET_CONTEXT_PATH).equals(Constants.ARCHIVE_ENTITY)) {
```
Comment (martind260) [inline src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java:50]: Das kann man in eine Methode "logException" auslagern.

File: src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java [hunk 3]
```diff
@@ -50,9 +61,19 @@ public void configure() {
...
                         exchange.getMessage().setHeader(Exchange.HTTP_RESPONSE_CODE, res.getStatus());
+                        if (exchange.getIn().getHeader(Constants.CAMEL_SERVLET_CONTEXT_PATH).equals(Constants.ARCHIVE_ENTITY)) {
+                            log.error(
+                                    String.format("Archive clean up (%s): %s", exchange.getProperty(Constants.ARCHIVE_ENTITY, MobidamArchive.class).toString(),
+                                            res.getError()));
+                        }
                     } else {
                         Throwable exception = exchange.getProperty(Exchange.EXCEPTION_CAUGHT, Throwable.class);
-                        log.error("Error occurred in route", exception);
+                        if (exchange.getIn().getHeader(Constants.CAMEL_SERVLET_CONTEXT_PATH).equals(Constants.ARCHIVE_ENTITY)) {
```
Comment (martind260) [inline src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java:71]: Das kann man in eine Methode "logException" auslagern.

File: src/main/java/de/muenchen/mobidam/service/ArchiveService.java [hunk 4]
```diff
@@ -15,4 +16,12 @@ public MobidamArchive save(MobidamArchive entity) {
         return archiveRepository.saveAndFlush(entity);
     }
 
+    public Iterable<MobidamArchive> expired(MobidamArchive entity) {
```
Comment (martind260) [inline src/main/java/de/muenchen/mobidam/service/ArchiveService.java:19]: getExpired oder listExpired

File: src/test/java/de/muenchen/mobidam/s3/S3ArchiveDeleteTest.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+    private static final String OBJECT_KEY = "File_1.csv";
+
+    private static final String PATH = "sub1/sub2/";
+
+    @Value("${mobidam.archive.name:archive}")
+    private String archive;
+
+    @Value("${mobidam.archive.expiration-months:1}")
+    private int expiration;
```
Comment (martind260) [inline src/test/java/de/muenchen/mobidam/s3/S3ArchiveDeleteTest.java:78]: var ist unbenutzt.

File: src/main/java/de/muenchen/mobidam/service/ArchiveService.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (sfi2022) [reply src/main/java/de/muenchen/mobidam/service/ArchiveService.java:19]: ✅

File: src/test/java/de/muenchen/mobidam/s3/S3ArchiveDeleteTest.java (hunk 5 above)
Comment (sfi2022) [reply src/test/java/de/muenchen/mobidam/s3/S3ArchiveDeleteTest.java:78]: ✅

File: src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java (hunk 1 above)
Comment (sfi2022) [reply src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java:37]: ✅

File: src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java (hunk 3 above)
Comment (sfi2022) [reply src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java]: ✅

File: src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java (hunk 2 above)
Comment (sfi2022) [reply src/main/java/de/muenchen/mobidam/s3/S3RouteBuilder.java]: ✅

**Review by martind260 — APPROVED** [review]

**Pull request merged**